runtime/packages/core-modules/assembly/executive.ts

```diff
@@ -47,11 +55,100 @@ export namespace Executive{
     export function finalizeBlock(): Header {
         return System.finalize();
     }
+    /**
+     * creates inherents from internal modules
+     * @param data inherents
+     */
+    export function createExtrinsics(data: InherentData): u8[] {
+        const timestamp: Inherent = Timestamp.createInherent(data);
+        const aura = AuraModule.createInherent(data);
+        if(aura.isSome()){
+            return System.ALL_MODULES.concat(timestamp.toU8a()).concat((<Inherent>aura.unwrap()).toU8a());
+        }
+        return System.ALL_MODULES.concat(timestamp.toU8a());
+    }
+
+    /**
+     * Apply Extrinsics
+     * @param ext extrinsic
+     */
+    export function applyExtrinsic(ext: u8[]): u8[] {
+        const encodedLen = Bytes.decodeCompactInt(ext);
+        return Executive.applyExtrinsicWithLen(ext, encodedLen.value as u32);
+    }
+    
+    /**
+     * 
+     * @param ext extrinsic
+     * @param encodedLen length
+     * @param encoded encoded extrinsic
+     */
+    export function applyExtrinsicWithLen(ext: u8[], encodedLen: u32): u8[]{
+        // if the length of the encoded extrinsic is less than 144, it's inherent
+        // in our case, we have only timestamp inherent
+        if(Inherent.isInherent(ext)){
+            const inherent = Inherent.fromU8Array(ext).result;
+            Timestamp.applyInherent(inherent);
+            return ApplyExtrinsicResult.SUCCESS;
+        }
+        const cmpLen = Bytes.decodeCompactInt(ext);
+        ext = ext.slice(cmpLen.decBytes);
+        const result = ext.shift();
+        if (result as bool){
+            const extrinsic = Extrinsic.fromU8Array(ext).result;
+            BalancesModule.applyExtrinsic(extrinsic);
+            return ApplyExtrinsicResult.SUCCESS;
+        }
+        return ApplyExtrinsicResult.CALL_ERROR;
+    }
+
+    /**
+     * Initial transaction validation
+     * @param source source of the transaction (external, inblock, etc.)
+     * @param utx transaction
+     */
+    export function validateTransaction(source: u8[], utx: Extrinsic): u8[] {
+        const from: AccountId = AccountId.fromU8Array(utx.from.toU8a()).result;
+        const fromBalance = BalancesModule.getAccountData(from);
+        const transfer = utx.getTransferBytes();
+
+        if(!Crypto.verifySignature(utx.signature, transfer, from)){
+            Log.error("Validation error: Invalid signature");
+            return TransactionError.INVALID_SIGNATURE;
+        }   
+        const nonce = System.accountNonce(from);
+        if (<u64>nonce.value >= <u64>utx.nonce.value){
+            Log.error("Validation error: Nonce value is less than or equal to the latest nonce");
+            return TransactionError.NONCE_TOO_LOW;
+        }
+        const balance: UInt128 = fromBalance.getFree();
+        if(!BalancesModule.validateTransaction(utx)){
+            Log.error("Validation error: Sender does not have enough balance");
```

Comment:
I was thinking that his log message is kind of coupling this `validateTransaction` function to the `BalancesModule.validateTransaction`. Specifying that the sender does not have enough balance implies that this function knows what checks are being performed in the `balances.validateTransaction` function. I think that it would be better if the Balances Module return the response code by itself and logs the message as-well.
One way to do it is to have `balances.validateTransaction` to return an object:
```JavaScript
{
    valid: bool,
    error: TransactionError,
    message: String
}
```
and in this `executive` module to do:
```JavaScript
const validation = BalancesModule.validateTransaction(utx)
if (!validation.valid) {
    Log.error(validation.message);
    return validation.error;
}
```